doc/develop/debug/index.rst

```diff
@@ -286,6 +286,81 @@ Support for Zephyr RTOS awareness is implemented in `pyOCD v0.11.0`_ and later.
 It is compatible with GDB PyOCD Debugging in Eclipse, but you must enable
 CONFIG_DEBUG_THREAD_INFO=y in your application.
 
+Debugging I2C communication
+***************************
+
+There is a possibility to log all or some of the I2C transactions done by application.
+This feature is enabled by the Kconfig option :kconfig:option:`CONFIG_I2C_DUMP_MESSAGES`, but it
+uses the ``LOG_DBG`` function to print the contents so also the I2C logging level has to be set
+to the debug level. It can be done by enabling the :kconfig:option:`CONFIG_I2C_LOG_LEVEL_DBG` option.
+
+The sample output of the dump looks like this::
+
+   D: I2C msg: io_i2c_ctrl7_port0, addr=50
+   D:    W      len=01: 00
+   D:    R Sr P len=08:
+   D: contents:
+   D: 43 42 41 00 00 00 00 00 |CBA.....
+
+The first line shows that the log belongs to the I2C transactions and informs which I2C interface
+was used to sent it and what is the address of the device targeted.
+In above example, the I2C interface is named ``io_i2c_ctrl7_port0`` and the target device address
+is ``0x50``
+
+.. note::
+
+   the address, length and contents values are in hexadecimal, but lack the ``0x`` prefix
+
+Next lines contain messages, both sent and received, and content of read messages being dumped
+depending on the parameter of the function ``i2c_dump_msgs_rw``. This function is available for use
+by user, but is also called internally by ``i2c_transfer`` API function with content dump enabled.
+Before the length parameter, the header of the message is printed using abbreviations:
+
+  - W - write message
+  - R - read message
+  - Sr - restart bit
+  - P - stop bit
+
+The above example shows one write message with byte ``0x00`` representing the address of register to
+read from the I2C target. After that the log shows the length of received message and following
+that, the bytes read from the target ``43 42 41 00 00 00 00 00``.
+The content dump consist of both the hex and ASCII representation.
+
+Filtering the I2C communication dump
+====================================
+
+During the development, it may be needed to investigate communication with one or more target
+devices. By default, all I2C communication is logged with every device on the bus.
+It may litter the log with unrelated devices and make it impossible to effectively debug the
+communication with device of interest.
+To resolve this issue, there is a feature that allows to select with which devices the communication
+should be logged.
+
+To enable this feature a Kconfig option :kconfig:option:`CONFIG_I2C_DUMP_MESSAGES_FILTER`
+must be enabled. The devices are selected using the devicetree node, for example::
```

Comment:
```suggestion
By default, all I2C communication is logged between all I2C controllers and I2C targets.
It may litter the log with unrelated devices and make it difficult to effectively debug the
communication with a device of interest.

Enable the Kconfig option :kconfig:option:`CONFIG_I2C_DUMP_MESSAGES_FILTER` to create an allowlist of I2C targets to log. The allowlist of devices is configured using the devicetree, for example::
```